TestRunnerWKWebView should simulate OS drag&drop events

squelart opened the pull request:
#### 5c22c9211fd848c9fb2cc053d7e672d135c1899b
<pre>
TestRunnerWKWebView should simulate OS drag&amp;drop events
<a href="https://bugs.webkit.org/show_bug.cgi?id=68552">https://bugs.webkit.org/show_bug.cgi?id=68552</a>
<a href="https://rdar.apple.com/158888281">rdar://158888281</a>

Reviewed by Wenson Hsieh and Ryosuke Niwa.

Expand [TestRunnerWKWebView dragImage:...] to initialize a
WebKitTestRunnerDraggingInfo with an incrementing sequence number; and
handle later mouse-down/dragged/up to generate corresponding drag&amp;drop
messages that the OS would normally send following real mouse events.

This makes a number of drag&amp;drop tests now work on WK2.

* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/mac-wk2/editing/pasteboard/dataTransfer-setData-getData-expected.txt: Removed.
* Tools/WebKitTestRunner/cocoa/TestRunnerWKWebView.mm:
(-[TestRunnerWKWebView dragImage:at:offset:event:pasteboard:source:slideBack:]):
(-[TestRunnerWKWebView mouseDown:]):
(-[TestRunnerWKWebView mouseUp:]):
(-[TestRunnerWKWebView mouseDragged:]):
* Tools/WebKitTestRunner/mac/WebKitTestRunnerDraggingInfo.h:
* Tools/WebKitTestRunner/mac/WebKitTestRunnerDraggingInfo.mm:
(-[WebKitTestRunnerDraggingInfo draggingSequenceNumber]):

Canonical link: <a href="https://commits.webkit.org/304228@main">https://commits.webkit.org/304228@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ec2ff8f96e6c079da048c98d67aaf4904c891708

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/134909 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/7328 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/46150 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/142416 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/86776 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/05416086-8296-4d74-87cb-3db850f44b19) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/136778 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/7948 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/7175 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/103079 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/70361 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/b5a207ce-b1eb-44e4-aa9e-d91e78153cb4) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/137855 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5615 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/120914 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83927 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/f710736f-c1f1-4a25-b760-82d90cbf1cb9) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5440 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/3058 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/3010 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/114656 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/39070 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/145115 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/7004 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/39645 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/111469 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/7058 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/5877 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111812 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28383 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/5275 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/117197 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/60928 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/7052 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/35370 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/6829 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/7066 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/6939 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Tools/WebKitTestRunner/cocoa/TestRunnerWKWebView.h:
```diff
@@ -28,12 +28,16 @@
-@interface TestRunnerWKWebView : WKWebView
+@interface TestRunnerWKWebView : WKWebView {
+@private
+    int _draggingSequenceNumber;
```
Nit - ivars should go in the `@implementation` when possible

pxlcoder commented on Tools/WebKitTestRunner/cocoa/TestRunnerWKWebView.h:
```diff
@@ -94,6 +98,7 @@
 @property (nonatomic, copy) void (^didHideContactPickerCallback)(void);
 @property (nonatomic, retain, setter=_setStableStateOverride:) NSNumber *_stableStateOverride;
 @property (nonatomic, setter=_setScrollingUpdatesDisabledForTesting:) BOOL _scrollingUpdatesDisabledForTesting;
+@property (nonatomic, retain) WebKitTestRunnerDraggingInfo *_currentDraggingInfo;
```
This doesn't appear to be used outside `TestRunnerWKWebView.mm`, so I think it could just be an ivar?

squelart committed:
TestRunnerWKWebView should simulate OS drag&drop events
https://bugs.webkit.org/show_bug.cgi?id=68552
rdar://158888281

Reviewed by Wenson Hsieh and Ryosuke Niwa.

Expand [TestRunnerWKWebView dragImage:...] to initialize a
WebKitTestRunnerDraggingInfo with an incrementing sequence number; and
handle later mouse-down/dragged/up to generate corresponding drag&drop
messages that the OS would normally send following real mouse events.

This makes a number of drag&drop tests now work on WK2.

* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/mac-wk2/editing/pasteboard/dataTransfer-setData-getData-expected.txt: Removed.
* Tools/WebKitTestRunner/cocoa/TestRunnerWKWebView.mm:
(-[TestRunnerWKWebView dragImage:at:offset:event:pasteboard:source:slideBack:]):
(-[TestRunnerWKWebView mouseDown:]):
(-[TestRunnerWKWebView mouseUp:]):
(-[TestRunnerWKWebView mouseDragged:]):
* Tools/WebKitTestRunner/mac/WebKitTestRunnerDraggingInfo.h:
* Tools/WebKitTestRunner/mac/WebKitTestRunnerDraggingInfo.mm:
(-[WebKitTestRunnerDraggingInfo draggingSequenceNumber]):

Canonical link: https://commits.webkit.org/304228@main
